airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/java/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.java

```diff
@@ -273,21 +274,27 @@ public void teardown() throws Exception {
     teardownNamespace(namespace);
   }
 
+  private DialectTableDefinition getFinalTableDefinition(List<StreamConfig> streamConfigs) throws Exception {
```

Comment:
it looks like you always call this method with a singleton list - could this just be `getFinalTableDefinition(StreamConfig stream)`, and then inside this method call `destinationHandler.gatherInitialState(List.of(stream));`?